integration_tests/grpc_rest/dclupgrade/helpers.go

```diff
@@ -0,0 +1,332 @@
+	}
+
+	return &res, nil
+}
+
+func GetProposedUpgrades(suite *utils.TestSuite) (res []dclupgradetypes.ProposedUpgrade, err error) {
+	if suite.Rest {
+		var resp dclupgradetypes.QueryAllProposedUpgradeResponse
+		err := suite.QueryREST("/dcl/dclupgrade/proposed-upgrades", &resp)
+		if err != nil {
+			return nil, err
+		}
+		res = resp.GetProposedUpgrade()
+	} else {
+		grpcConn := suite.GetGRPCConn()
+		defer grpcConn.Close()
+
+		// This creates a gRPC client to query the x/dclauth service.
+		dclupgradeClient := dclupgradetypes.NewQueryClient(grpcConn)
+		resp, err := dclupgradeClient.ProposedUpgradeAll(
+			context.Background(),
+			&dclupgradetypes.QueryAllProposedUpgradeRequest{},
+		)
+		if err != nil {
+			return nil, err
+		}
+		res = resp.GetProposedUpgrade()
+	}
+
+	return res, nil
+}
+
+func GetApprovedUpgrades(suite *utils.TestSuite) (res []dclupgradetypes.ApprovedUpgrade, err error) {
+	if suite.Rest {
+		var resp dclupgradetypes.QueryAllApprovedUpgradeResponse
+		err := suite.QueryREST("/dcl/dclupgrade/approved-upgrades", &resp)
+		if err != nil {
+			return nil, err
+		}
+		res = resp.GetApprovedUpgrade()
+	} else {
+		grpcConn := suite.GetGRPCConn()
+		defer grpcConn.Close()
+
+		// This creates a gRPC client to query the x/dclauth service.
+		dclupgradeClient := dclupgradetypes.NewQueryClient(grpcConn)
+		resp, err := dclupgradeClient.ApprovedUpgradeAll(
+			context.Background(),
+			&dclupgradetypes.QueryAllApprovedUpgradeRequest{},
+		)
+		if err != nil {
+			return nil, err
+		}
+		res = resp.GetApprovedUpgrade()
+	}
+
+	return res, nil
+}
+
+func DCLUpgradeDemo(suite *utils.TestSuite) {
+	// Alice and Bob are predefined Trustees
+	aliceName := testconstants.AliceAccount
+	aliceKeyInfo, err := suite.Kr.Key(aliceName)
+	require.NoError(suite.T, err)
+	aliceAccount, err := test_dclauth.GetAccount(suite, aliceKeyInfo.GetAddress())
+	require.NoError(suite.T, err)
+
+	bobName := testconstants.BobAccount
+	bobKeyInfo, err := suite.Kr.Key(bobName)
+	require.NoError(suite.T, err)
+	bobAccount, err := test_dclauth.GetAccount(suite, bobKeyInfo.GetAddress())
+	require.NoError(suite.T, err)
+
+	// trustee proposes upgrade
+	proposeUpgradeMsg := NewMsgProposeUpgrade(aliceAccount.Address, utils.RandString(), 100000, utils.RandString())
+	_, err = suite.BuildAndBroadcastTx([]sdk.Msg{proposeUpgradeMsg}, aliceName, aliceAccount)
+	require.NoError(suite.T, err)
+
+	// Check upgrade is proposed
+	proposedUpgrade, err := GetProposedUpgrade(suite, proposeUpgradeMsg.Plan.Name)
+	require.NoError(suite.T, err)
+	require.Equal(suite.T, proposeUpgradeMsg.Creator, proposedUpgrade.Creator)
+	require.Equal(suite.T, proposeUpgradeMsg.Plan, proposedUpgrade.Plan)
+
+	// Get all proposed upgrades
+	_, err = GetProposedUpgrades(suite)
+	require.NoError(suite.T, err)
+
+	// Get all approved upgrades
+	approvedUpgrades, err := GetApprovedUpgrades(suite)
+	require.NoError(suite.T, err)
+	require.NotContains(suite.T, approvedUpgrades, proposedUpgrade)
```

Comment:
At first, `proposedUpgrade` is a pointer. But even with using `*proposedUpgrade`, this check would not make sense. It would always succeed because `*proposedUpgrade` type and `approvedUpgrades` element type are different.

To perform such the check, the value of `*proposedUpgrade` should be converted to `ApprovedUpgrade` at the beginning. But it will be simpler just to check the approved upgrade presence by a name using `GetApprovedUpgrade`.

Reply:
fixed